### PR TITLE
Add a device to mount the host /dev/shm inside the container.

### PR DESCRIPTION
**Pull request by jschwartzentruber**
We do this when fuzzing, since Firefox needs a lot of space in /dev/shm and sizing it properly is not straightforward.

### CODE REVIEW

File: src/lib/devices/shared_memory_device_manager.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    return device;
+  }
+
+  getAvailableDevices() {
+    return this.devices;
+  }
+}
+
+class SharedMemoryDevice {
+  constructor(path) {
```
Comment (walac) [inline src/lib/devices/shared_memory_device_manager.js]: Why does the constructor accept a `path` parameter if only `/dev/shm` is allowed?

**Comment by walac** [comment]
I create a [PR](https://github.com/mozilla/community-tc-config/pull/174) in the community repo to add the shared memory capability scope.

**Comment by walac** [comment]
There is something wrong with the linter task, but I am quite sure it is unrelated to your patch. I am going to investigate it.

**Comment by walac** [comment]
Ok, I found out what's wrong. As you added the `hostSharedMemory` to `.taskcluster.yml` this created a chicken-egg problem, since the currently deployed docker-worker obvisiously doesn't support this device. I wonder how tests were passing before.

**Comment by jschwartzentruber** [comment]
How do we get around this chicken-egg problem? Is there anything for me to do?

**Pull request merged**